apps/withdraw-pool/src/app/signing/page.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+
+import { SignatureSteps } from "#/components/SignaturesSteps";
+import { WaitingSignature } from "#/components/WaitingSignature";
+import { useIFrameContext } from "#/context/iframe";
+import { useCowShedSignature } from "#/hooks/useCowShedSignature";
+import { useHandleTokenAllowance } from "#/hooks/useHandleTokenAllowance";
+import { useSubmitHook } from "#/hooks/useSubmitHook";
+import { BaseTransaction } from "#/utils/transactionFactory/types";
+import { BigNumber, BigNumberish } from "ethers";
+import { useCallback, useMemo, useState } from "react";
+import { Address } from "viem";
+
+export default function Page() {
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const [permitTxs, setPermitTxs] = useState<BaseTransaction[]>([]);
+  const { hookInfo, cowShed } = useIFrameContext();
+  const submitHook = useSubmitHook();
+  const cowShedSignature = useCowShedSignature();
+  const handleTokenAllowance = useHandleTokenAllowance();
+
+  const cowShedCallback = useCallback(async () => {
+    if (!cowShedSignature || !hookInfo || !cowShed) return;
+
+    const txs = [...permitTxs, ...hookInfo.txs];
+
+    const cowShedCall = await cowShedSignature(txs);
+    if (!cowShedCall) throw new Error("Error signing hooks");
+    submitHook({
+      target: cowShed.getFactoryAddress(),
+      callData: cowShedCall,
+    });
+  }, [cowShedSignature, hookInfo, permitTxs, cowShed]);
+
+  const permitCallback = useCallback(
+    async (permit: {
+      tokenAddress: string;
+      amount: BigNumberish;
+      tokenSymbol: string;
+    }) => {
+      const permitData = await handleTokenAllowance(
+        BigNumber.from(permit.amount),
+        permit.tokenAddress as Address
+      );
+
+      if (permitData) {
+        setPermitTxs((prev) => [
+          ...prev,
+          {
+            to: permitData.target,
+            value: BigInt(0),
+            callData: permitData.callData,
+          },
+        ]);
+      }
+      setCurrentIndex((prev) => prev + 1);
```

Comment:
What do you think of changing this prev to another name, like "i", that we commonly use for index?